src/common/utils.py

```diff
@@ -241,20 +241,45 @@ def _read_cfnconfig():
     return cfnconfig_params
 
 
-def _get_instance_info_from_pricing_file(region, proxy_config, instance_type):
+@retry(
+    stop_max_attempt_number=5,
+    wait_exponential_multiplier=5000,
+    retry_on_exception=lambda exception: isinstance(exception, ClientError)
+    and exception.response.get("Error").get("Code") == "RequestLimitExceeded",
+)
+def _fetch_instance_info(region, proxy_config, instance_type):
     """
-    Read pricing file and get number of vcpus and gpus for the given instance type.
+    Fetch instance type information from EC2's DescribeInstanceTypes API.
 
-    :return: (the number of vcpus or -1 if the instance type cannot be found,
-                number of gpus or None if the instance does not have gpu)
+    :param region: AWS region
+    :param proxy_config: proxy configuration
+    :param instance_type: instance type to get info for
+    :return: dict with the same format as those returned in the "InstanceTypes" array of the
+             object returned by DescribeInstanceTypes
     """
-    instances = _fetch_pricing_file(region, proxy_config)
-    if instance_type not in instances:
-        error_msg = "Instance type {0} not found in instances.json file.".format(instance_type)
+    ec2_client = boto3.client("ec2", region_name=region, config=proxy_config)
+    try:
+        return ec2_client.describe_instance_types(InstanceTypes=[instance_type]).get("InstanceTypes")[0]
+    except (ClientError, Exception) as exception:
+        error_msg = "Error when calling DescribeInstanceTypes for instance type {0}: {1}".format(
+            instance_type,
+            exception.response.get("Error").get("Message") if isinstance(exception, ClientError) else exception,
+        )
         log.critical(error_msg)
         raise CriticalError(error_msg)
```

Comment:
if we always raise CriticalError on ClientError I'm afraid the retry is never applied

Reply:
good call, i missed that

Reply:
should be addressed in the changes i just pushed